qiita_db/analysis.py

```diff
@@ -160,6 +162,22 @@ def create(cls, owner, name, description, parent=None, from_default=False):
             conn_handler.add_to_queue(
                 queue, sql, (owner.id, name, description, status_id))
 
+        # make changes to SQL needed to add to both QIITA and given portal
+        insert_portal = ""
```

Comment:
This entire block of code can be replaced by:
```python
sql = """INSERT INTO qiita.analysis_portal (analysis_id, portal_type_id)
         VALUES (%s, %s)"""
args = [('{0}', portal_id)]
if qiita_config.portal != 'QIITA':
    qp_id = convert_to_id('QIITA', 'portal_type', 'portal')
    args.append(('{0}', qp_id))
conn_handler.add_to_queue(queue, sql, args, many=True)
```
which is easy to understand and maintain.